components/Swipe/index.js

```diff
@@ -0,0 +1,241 @@
+import React, { Component, PropTypes } from 'react';
+import classNames from 'classnames';
+import './app.css';
+
+function autoBind(methods, context) {
+  methods.forEach(method => {
+    // eslint-disable-next-line no-param-reassign
+    context[method] = context[method].bind(context);
+  });
+}
+
+export default class Swipe extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentIndex: props.initialIndex,
+      width: null,
+      scrollLeft: 0,
+      drag: 0
+    };
+
+    autoBind([
+      'getImages',
+      'setWidth',
+      'handleTouchStart',
+      'handleTouchEnd',
+      'handleTouchMove',
+      'gotoNext',
+      'gotoPrev',
+      'onChange',
+      'lazyLoadImage',
+      'initLazyLoad'
+    ], this);
+  }
+
+  componentDidMount() {
+    this.setWidth();
+    this.initLazyLoad();
+
+    window.addEventListener('resize', this.setWidth);
+  }
+
+  setWidth() {
+    if (this.swipeRef) {
+      this.setState({
+        width: this.swipeRef.clientWidth
```

Comment:
This property will round the value to an integer. If you need a fractional value, use element.getBoundingClientRect().